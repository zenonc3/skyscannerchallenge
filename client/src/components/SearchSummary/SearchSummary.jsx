import PropTypes from 'prop-types';
import BpkText from 'bpk-component-text/src/BpkText';
import React from 'react';
import STYLES from './SearchSummary.scss';

const c = className => STYLES[className] || 'UNKNOWN';

const SearchSummary = ({
  origin, destination, numTravellers, cabinClass,
}) =>
  (
    <div className={c('SearchSummary__container')}>
      <div>
        <BpkText textStyle="xl">{origin}</BpkText>
        <BpkText textStyle="xl"> â†’ </BpkText>
        <BpkText textStyle="xl">{destination}</BpkText>
      </div>
      <div>
        <BpkText textStyle="sm">{numTravellers} traveller(s), {cabinClass}</BpkText>
      </div>
    </div>
  );

SearchSummary.propTypes = {
  origin: PropTypes.string.isRequired,
  destination: PropTypes.string.isRequired,
  numTravellers: PropTypes.number.isRequired,
  cabinClass: PropTypes.string.isRequired,
};

export default SearchSummary;
