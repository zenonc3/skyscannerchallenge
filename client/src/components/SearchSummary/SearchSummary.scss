@import '~bpk-mixins';

.SearchSummary {
  &_container {
    padding: $bpk-spacing-sm;
    width: 100%;
    background-color: rgb(0, 178, 212);
    color: white;
  }
}
