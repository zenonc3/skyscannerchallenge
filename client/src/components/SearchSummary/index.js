import SearchSummary from './SearchSummary';

export default SearchSummary;
