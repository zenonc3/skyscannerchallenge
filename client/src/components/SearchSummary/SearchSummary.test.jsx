import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SearchSummary from './SearchSummary';

describe('Search Summary', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<SearchSummary origin="a" destination="b" cabinClass="eco" numTravellers={2} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

