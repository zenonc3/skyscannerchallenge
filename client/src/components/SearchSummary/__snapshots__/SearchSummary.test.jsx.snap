// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Summary should render correctly 1`] = `
<div
  className="SearchSummary__container"
>
  <div>
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="xl"
    >
      a
    </BpkText>
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="xl"
    >
       â†’ 
    </BpkText>
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="xl"
    >
      b
    </BpkText>
  </div>
  <div>
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="sm"
    >
      2
       traveller(s), 
      eco
    </BpkText>
  </div>
</div>
`;
